### PR TITLE
template add cloud init

### LINKED ISSUE #55342: [RFE] Extend  ovirt_template Module - VM Definitions
##### SUMMARY

The **ovirt_template** module needs to be able to provide and set multiple features similar to the **ovirt_vm**.

Need to be able to set multiple pieces of information (USB Redirect, Networking/vNICS, Cloud-Init, etc.). Essentially, this is missing the rest of the features that can be specified from the RHVM UI.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

ovirt_template

Pieces to add are from the **ovirt_vm** module.

vnic_profile_mappings
usb_support
timezone
sso
soundcard_enabled
smartcard_enabled
nics
memory_max
memory
memory_guaranteed
cloud_init_persist
cloud_init_nics
cloud_init
ballooning_enabled

##### ADDITIONAL INFORMATION

This feature would provide the full functionality for the **ovirt_template** module allowing items to be changed or set differently than the VM that the template was based on. Usage would be similar to that of the **ovirt_vm** module.

```yaml

```

### PR DESCRIPTION
**Pull request by mnecas**
##### SUMMARY

ovirt template add 

usb_support
timezone
sso
soundcard_enabled
smartcard_enabled
nics
cloud_init_persist
cloud_init_nics
cloud_init
ballooning_enabled

Fixes #55342

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

ovirt
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by mnecas** [comment]
@tmichett I added those in summary rest was already implemented.

**Comment by mnecas** [comment]
@machacekondra should i also add sysprep?

File: lib/ansible/modules/cloud/ovirt/ovirt_template.py [hunk 1]
```diff
@@ -189,6 +189,147 @@
...
+                    - Type of the network interface.
+                choices: ['virtio', 'e1000', 'rtl8139']
+                default: 'virtio'
+            mac_address:
+                description:
+                    - Custom MAC address of the network interface, by default it's obtained from MAC pool.
+                    - "NOTE - This parameter is used only when C(state) is I(running) or I(present) and is able to only create NICs.
+                    To manage NICs of the VM in more depth please use M(ovirt_nics) module instead."
+        version_added: "2.9"
+    cloud_init_persist:
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_template.py]: It doesn't make sense in case of template as there is no run-once, so plaese remove it.

**Comment by machacekondra** [comment]
> @machacekondra should i also add sysprep?

Yes please

**Comment by machacekondra** [comment]
shipit

**Pull request merged**